.github/workflows/pull_request.yml

```diff
@@ -0,0 +1,55 @@
+name: Pull Request
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+    - source/**
+    - clik/**
+    - modules/**
+    - examples/**
+    - cmake/**
+    - hal/**
+    - .github/actions/do_build_ock/**
+    - .github/actions/setup_ubuntu_build/**
+    - .github/workflows/pull_request.yml
+    - CMakeLists.txt
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+jobs:
+
+############### JOB   mr-ubuntu-gcc-x86_64-riscv-fp16-cl3-0-unitcl_vecz:
+
+  mr-ubuntu-gcc-x86_64-riscv-fp16-cl3-0-unitcl_vecz:
+    runs-on: ubuntu-22.04
+    timeout-minutes: 60
+    steps:
+    - name: Checkout repo
+      uses: actions/checkout@v4.1.0
+      with:
+        path: code
```

Comment:
I still think I'd need convincing on this one.

Reply:
Sure. This PR's job doesn't show all the issues that well. [Here's another example in Gitlab](https://git.office.codeplay.com/ComputeAorta/ci-github/-/blob/main/.gitlab/ci/merge-request.yml?ref_type=heads#L496)
... note the `--source_dir` option to `build.py`.
I guess we're looking for a consistent structure that works across all jobs when `build.py` is removed from each. And possibly something that tidies up from the Gitlab offering too as it can be tricky to follow. Any improvement on my offering is fine by me.

Reply:
that still doesn't really convince me, the checkout to code and copy of .github feels a bit clumsy to me. I don't see this approach on other github workflows elsewhere. I'd welcome @hvdijk's opinion though.

Reply:
As discussed off-line - I will refactor to restore the Gitlab model.

Reply:
Done